clang/include/clang/Basic/AttributeCommonInfo.h

```diff
@@ -152,7 +152,12 @@ class AttributeCommonInfo {
     // FIXME: Eventually we want to do a list here populated via tablegen.  But
     // we want C++ attributes to be permissible on Lambdas, and get propagated
     // to the call operator declaration.
-    return getParsedKind() == AT_SYCLIntelKernelArgsRestrict;
+    auto ParsedAttr = getParsedKind();
+    if (ParsedAttr == AT_SYCLIntelKernelArgsRestrict ||
```

Comment:
Just as a note, I understand that some conflicts will need to be resolved either here or for #871, depending on which gets in first.

Reply:
Yes, it's expected.